src/materials/Material.js

```diff
@@ -86,6 +86,15 @@ Object.assign( Material.prototype, EventDispatcher.prototype, {
 
 			}
 
+			// for backward compatability if shading is set in the constructor
+			if ( key === 'shading' ) {
+
+				console.warn( 'THREE.' + this.type + ': .shading has been removed. Use the boolean .flatShading instead.' );
+				this.flatShading = ( newValue === FlatShading ) ? true : false;
```

Comment:
There are other properties for which the `Legacy` warnings are not effective when the property is set in the constructor -- maybe a lot of them...

Do you prefer constructors with no args -- i.e., modifying the default properties after instantiation?